File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int temp=x;
        long reverse=0;
        while(temp>0)
        {
            int rem=temp%10;
            reverse=(reverse*10)+rem;
            temp/=10;
        }
        if(x==reverse){
            return true;
        }
        else{
            return false;
        }
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
public class Solution {

    public boolean isPalindrome(int n) {
        if (n < 0) return false; // negative numbers are not palindromes

        int revNum = 0, dup = n;
        while (n > 0) {
            int ld = n % 10;
            revNum = revNum * 10 + ld;
            n /= 10;
        }
        return dup == revNum;
    }
}
